### ISSUE TITLE
scenes consistently swallowing keyboard events after switching back and forth

### ISSUE THREAD

**Issue by andybarron** [excaliburjs/Excalibur#3126]
NB: Thanks so much for this incredible project! I'm happy to take a stab at fixing this if anyone can point me in the right direction :)

### Steps to Reproduce

Create two scenes that toggle back and forth with **two different key presses**:

```ts
import * as ex from "excalibur";

const a = new ex.Scene();
a.on("initialize", () => {
  a.input.keyboard.on("press", (e) => {
    if (e.key === ex.Keys.Key1) {
      a.engine.goToScene("b");
    }
  });
});

const b = new ex.Scene();
b.backgroundColor = ex.Color.DarkGray;
b.on("initialize", () => {
  b.input.keyboard.on("press", (e) => {
    if (e.key === ex.Keys.Key2) {
      b.engine.goToScene("a");
    }
  });
});

const game = new ex.Engine({
  scenes: { a, b },
});

game.goToScene("a");
game.start();
```

The issue does not appear if you target the same key code in both scenes.

### Expected Result

* Pressing "1" always transitions scene A to scene B
* Pressing "2" always transitions scene B to scene A

### Actual Result

* Pressing "1" always transitions scene A to scene B ✅ 
* Pressing "2" always transitions scene B to scene A ❌ 
  * It works once. **On subsequent attempts, you have to press "2" twice**.
  * It seems like the key gets "stuck" in the down state due to the scene changes, and the first press "releases" it.
  * Not sure why it doesn't happen to the initial scene.

### Environment

- browsers and versions: Firefox (128.0, 64-bit), Chrome (126.0.6478.127, arm64)
- operating system: macOS 14.5 (Sonoma) - MacBook Air (M1, 2020)
- Excalibur versions: 0.29.0, 0.29.3, 0.30.0-alpha.1105
  - Older versions (<=28.0) don't have the scene-specific input API, so they seem to be unaffected

### Current Workaround

Haven't found one 😞

**Comment by eonarheim** [comment]
@andybarron Thanks for the issue, and the kind words! 

Definitely feels like a bug to me, let me dig around quick with your example to see if anything jumps out.

**Comment by eonarheim** [comment]
@andybarron This is merged into main, should be deployed to the alpha npm package soon. 

I have a few more bugs to fix before I cut the next v0.29.4, probably end of July?

**Comment by andybarron** [comment]
@eonarheim just installed the latest alpha and it works like a charm! thanks 😄 🙏🏻
